include/wlr/xwayland.h

```diff
@@ -53,6 +63,16 @@ struct wlr_xwayland_surface {
 	xcb_atom_t *protocols;
 	size_t protocols_len;
 
+	uint32_t decorations;
+
+	#ifdef HAS_XCB_ICCCM
+	xcb_icccm_wm_hints_t *hints;
+	xcb_size_hints_t *size_hints;
+	#else
+	void *hints;
+	void *size_hints;
+	#endif
```

Comment:
You know, as I think about it more, it probably makes more sense to convert the xcb structure into a wlr structure and expose that type regardless of whether or not we have support for it. Then it can just always be NULL if we don't support it and users don't have to put their code accessing the hints behind ifdefs

Reply:
Yeah, true. This bugs me for two reasons:
- We'll have a mix of `xcb` and `wlr` data types
- Users won't be able to use functions like `xcb_icccm_wm_hints_get_urgency`, because they take `xcb_icccm_wm_hints_t` as an argument

That said, we can probably call these for the user, and put the result in our struct.

Reply:
Aye, we can do just that.

Reply:
I ended up just copying these two structs into wlroots, and adding an extra field for urgency.